File: app/src/main/java/net/jkcat/jkit/java/MainActivity.java
package net.jkcat.jkit.java;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.google.gson.Gson;

import net.jkcat.jkit.R;
import net.jkcat.jkit.bean.ActivityBean;
import net.jkcat.jkit.java.activity.ActionDetailActivity;
import net.jkcat.jkit.java.list.ActionListActivity;
import net.jkcat.jkit.network.ApiService;
import net.jkcat.network.BaseObserver;

import org.jetbrains.annotations.NotNull;

import io.reactivex.disposables.Disposable;

/**
 * @author JokerCats on 2021.05.08
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_request).setOnClickListener(v -> getBanner());

        findViewById(R.id.btn_demo_activity).setOnClickListener(v ->
                startActivity(new Intent(this, ActionDetailActivity.class)));

        findViewById(R.id.btn_demo_list).setOnClickListener(v -> {
            startActivity(new Intent(this, ActionListActivity.class));
        });
    }

    private void getBanner() {
        JNetApi.getInstance().getService(ApiService.class)
                .getActivityInfo()
                .compose(JNetApi.getInstance().applySchedulers())
                .subscribe(new BaseObserver<ActivityBean>() {
                    @Override
                    public void onRequestInit(@NotNull Disposable disposable) {

                    }

                    @Override
                    public void onRequestFailure(@NotNull Throwable e) {
                        Log.i("net", "request error msg is ${e.message}");
                    }

                    @Override
                    public void onRequestSuccess(@NotNull ActivityBean result) {
                        Log.i("net", new Gson().toJson(result));
                    }
                });
    }
}

File: app/src/main/java/net/jkcat/jkit/kotlin/MainActivity.kt
package net.jkcat.jkit.kotlin

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.google.gson.Gson
import io.reactivex.disposables.Disposable
import kotlinx.android.synthetic.main.activity_main.*
import net.jkcat.jkit.R
import net.jkcat.jkit.bean.ActivityBean
import net.jkcat.jkit.network.ApiService
import net.jkcat.network.BaseObserver

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_request.setOnClickListener {
            getBanner()

//            UpdaterHelper.Builder(this)
//                .url("https://dl.yzcdn.cn/koudaitong.apk")
//                .title("发现新版本")
//                .content("这是更新的内容")
//                .name("测试更新")
//                .description("这是更新标题")
//                .force(false)
//                .create()
//                .launcher()
        }
    }

    private fun getBanner() {
        JNetApi.getService(ApiService::class.java)
            .getActivityInfo()
            .compose(JNetApi.applySchedulers())
            .subscribe(object : BaseObserver<ActivityBean>() {
                override fun onRequestSuccess(result: ActivityBean) {
                    Log.i("net", Gson().toJson(result))
                }

                override fun onRequestFailure(e: Throwable) {
                    Log.i("net", "request error msg is ${e.message}")
                }

                override fun onRequestInit(disposable: Disposable) {

                }
            })
    }

}